test/unit/test_core_dbt_utils.py

```diff
@@ -0,0 +1,41 @@
+import requests
+import unittest
+
+from dbt.exceptions import ConnectionException
+from dbt.utils import _connection_exception_retry as connection_exception_retry
+
+
+class testCoreDbtUtils(unittest.TestCase):
+    def test_connection_exception_retry_none(self):
+        Counter._reset()
+        connection_exception_retry(lambda: Counter._add(), 5)
+        self.assertEqual(1, counter)
+
+    def test_connection_exception_retry_max(self):
+        Counter._reset()
+        self.assertRaises(ConnectionException, connection_exception_retry(lambda: Counter._add_with_exception(), 5))
```

Comment:
This `assertRaises` doesn't work for me where it doesn't catch the exception (its the and the `ConnectionException` that is listed) and the test fails because an exception is hit. Can anyone see why that would be the case instead of it just succeeding that it was raised?